drivers/frequency/admfm2000/iio_admfm2000.c

```diff
@@ -0,0 +1,206 @@
+ *   @file   iio_admfm2000.c
+ *   @brief  Implementation of admfm2000 IIO Driver.
+ *   @author Ramona Nechita (ramona.nechita@analog.com)
+********************************************************************************
+ * Copyright 2025(c) Analog Devices, Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. “AS IS” AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL ANALOG DEVICES, INC. BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include "no_os_error.h"
+#include "no_os_util.h"
+#include "iio_admfm2000.h"
+#include "no_os_alloc.h"
+
+/**
+ * @brief Get the channel gain.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with the data to be written.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return 	  - The size of the read data in case of success, error code
+ * 		    otherwise.
+ */
+static int admfm2000_iio_read_gain(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel,
+				   intptr_t priv)
+{
+	struct admfm2000_iio_dev *iio_admfm2000 = (struct admfm2000_iio_dev *)dev;
+	struct admfm2000_dev *admfm2000_dev;
+	int32_t gain;
+	int32_t vals[2];
+	int32_t tmp;
+	int32_t ret;
+
+	if (!iio_admfm2000)
+		return -EINVAL;
+
+	admfm2000_dev = iio_admfm2000->admfm2000_dev;
+
+	if (!admfm2000_dev)
+		return -EINVAL;
+
+	ret = admfm2000_get_gain(admfm2000_dev, channel->ch_num, &gain);
```

Comment:
ups, was this tested?

Reply:
not after the name change, but it's fixed now